api/k0smotron.io/v1beta1/k0smotroncluster_types.go

```diff
@@ -65,6 +65,12 @@ type ClusterSpec struct {
 	//+kubebuilder:validation:Enum:=kuberouter;calico;custom
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="cniPlugin is immutable"
 	CNIPlugin string `json:"cniPlugin,omitempty"`
+	// Manifests allows to specify a confiMmap with manifests to be
+	// applied to the cluster. The configmap will be mounted as a volime
+	// in /var/lib/k0s/manifests. For more information check
+	// https://docs.k0sproject.io/latest/manifests/
+	//+kubebuilder:validation:Optional
+	Manifests string `json:"manifests,omitempty"`
```

Comment:
Could we use a more verbose name like `ManifestSecrets`?

Reply:
Instead I think the best would be to use a `[]v1.Volume`. I think this only has benefits:
- The name must be a dns label, which works for the filesystem and we don't need to define a name for the stack, just pick the volume name
- We allow the the user to define either a secret or a configmap (or something else, up to them)
- Pretty much every kubernetes intermediate user is already familiar with it.
- Is part of the stable API so it's not going anywhere any time soon.

I updated the PR with this. What do you think?

Reply:
Makes sense to me 👍